controllers/console_controller.go

```diff
@@ -0,0 +1,135 @@
+package controllers
+
+import (
+	"context"
+	"strings"
+
+	"github.com/go-logr/logr"
+	consolev1alpha1 "github.com/openshift/api/console/v1alpha1"
+	operatorv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	console "github.com/red-hat-data-services/odf-operator/console"
+	"github.com/red-hat-data-services/odf-operator/pkg/util"
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+type ConsoleReconciler struct {
+	client.Client
+	Log          logr.Logger
+	ConsoleImage string
+	ConsolePort  int
+}
+
+//+kubebuilder:rbac:groups="",resources=deployments,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=services,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=operators.coreos.com,resources=clusterserviceversions,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=operators.coreos.com,resources=subscriptions,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=console.openshift.io,resources=consoleplugins,verbs=*
+
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
+func (r *ConsoleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := r.Log.WithValues("odf-console", req.NamespacedName)
+	// Create deployment if doesn't exist
+	deployment := &appsv1.Deployment{}
+	idealDeployment := console.GetDeployment(r.ConsoleImage, r.ConsolePort)
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: idealDeployment.ObjectMeta.Name, Namespace: idealDeployment.ObjectMeta.Namespace}, deployment); errors.IsNotFound(err) {
+		if err := r.Client.Create(ctx, &idealDeployment); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
```

Comment:
Please check the below comment for `sevrice` variable which applicable for `deployment` var.